src/vuln_analysis/server/webhooks/envelope.py

```
@@ -0,0 +1,228 @@
+    if not decoded_path.startswith("/reports/"):
+        return None
+    if any(segment in (".", "..") for segment in decoded_path.split("/")):
+        return None
+
+    normalized_path = posixpath.normpath(decoded_path)
+    if not normalized_path.startswith("/reports/"):
+        return None
+    return quote(normalized_path, safe="/-._~")
+
+
+def _sanitize_result_paths(value: Any) -> Any:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `value`

(ANN401)